Modules/Quotas/src/Quotas.Application/Tiers/Queries/GetTierById/Handler.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using MediatR;
+
+namespace Backbone.Modules.Quotas.Application.Tiers.Queries.GetTierById;
+public class Handler : IRequestHandler<GetTierByIdQuery, GetTierByIdResponse>
+{
+    private readonly ITiersRepository _tiersRepository;
+    private readonly IMetricsRepository _metricsRepository;
+
+    public Handler(ITiersRepository tiersRepository, IMetricsRepository metricsRepository)
+    {
+        _tiersRepository = tiersRepository;
+        _metricsRepository = metricsRepository;
+    }
+
+    public async Task<GetTierByIdResponse> Handle(GetTierByIdQuery request, CancellationToken cancellationToken)
+    {
+        var tier = await _tiersRepository.Find(request.Id, cancellationToken);
+
+        var metricsKeys = tier.Quotas.DistinctBy(quota => quota.MetricKey).Select(quota => quota.MetricKey);
+        var metrics = metricsKeys.Select(metricKey => _metricsRepository.Find(metricKey, cancellationToken).Result);
```

Comment:
Why do you access the `Result` property? 😲 If the `Find` method is actually async one day, then this code would produce an error